src/main/java/bio/terra/service/load/flight/LoadLockStep.java

```diff
@@ -20,10 +22,14 @@ public LoadLockStep(LoadService loadService) {
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException {
     String loadTag = loadService.getLoadTag(context);
-    UUID loadId = loadService.lockLoad(loadTag, context.getFlightId());
-    FlightMap workingMap = context.getWorkingMap();
-    workingMap.put(LoadMapKeys.LOAD_ID, loadId.toString());
-    return StepResult.getStepResultSuccess();
+    try {
+      UUID loadId = loadService.lockLoad(loadTag, context.getFlightId());
+      FlightMap workingMap = context.getWorkingMap();
+      workingMap.put(LoadMapKeys.LOAD_ID, loadId.toString());
+      return StepResult.getStepResultSuccess();
+    } catch (LoadLockedException ex) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, ex);
+    }
   }
 
   @Override
```

Comment:
I want to double check.  I think you're right that we don't need a retry around the undo but just want to call out and make sure that we don't

Reply:
Thanks for checking!  You are correct, `loadService.unlockLoad` does not throw so I don't think we need retry handling here:

https://github.com/DataBiosphere/jade-data-repo/blob/8e4633494ac4ba82cb21273211bd261ac379dfe8/src/main/java/bio/terra/service/load/LoadDao.java#L116-L137

Reply:
right on.  That makes a ton of sense